java/core/src/test/java/sleeper/core/range/canonicaliser/RangeCanonicaliserByteArrayTypeTest.java

```diff
@@ -0,0 +1,124 @@
+        //Given
+        Range range = rangeFactory.createRange(field, new byte[]{1}, new byte[]{10});
+
+        //When
+        Range canonicalisedRange = RangeCanonicaliser.canonicaliseRange(range);
+
+        //Then
+        assertThat(range.isInCanonicalForm()).isTrue();
+        assertThat(canonicalisedRange).isEqualTo(range);
+    }
+
+    @Test
+    public void shouldCanonicaliseRangeMinInclusiveMaxInclusive() {
+        //Given
+        Range range = rangeFactory.createRange(field, new byte[]{1}, true, new byte[]{10}, true);
+
+        //When
+        Range canonicalisedRange = RangeCanonicaliser.canonicaliseRange(range);
+
+        //Then
+        assertThat(range.isInCanonicalForm()).isFalse();
+        assertThat(canonicalisedRange).isEqualTo(rangeFactory.createRange(field, new byte[]{1}, nextByteArrayValue(new byte[]{10})));
+    }
+
+    @Test
+    public void shouldCanonicaliseRangeMinExclusiveMaxInclusive() {
+        //Given
+        Range range = rangeFactory.createRange(field, new byte[]{1}, false, new byte[]{10}, true);
+
+        //When
+        Range canonicalisedRange = RangeCanonicaliser.canonicaliseRange(range);
+
+        //Then
+        assertThat(range.isInCanonicalForm()).isFalse();
+        assertThat(canonicalisedRange).isEqualTo(rangeFactory.createRange(field, nextByteArrayValue(new byte[]{1}), nextByteArrayValue(new byte[]{10})));
+    }
+
+    @Test
+    public void shouldCanonicaliseRangeMinExclusiveMaxExclusive() {
+        //Given
+        Range range = rangeFactory.createRange(field, new byte[]{1}, false, new byte[]{10}, false);
+
+        //When
+        Range canonicalisedRange = RangeCanonicaliser.canonicaliseRange(range);
+
+        //Then
+        assertThat(range.isInCanonicalForm()).isFalse();
+        assertThat(canonicalisedRange).isEqualTo(rangeFactory.createRange(field, nextByteArrayValue(new byte[]{1}), new byte[]{10}));
+    }
+
+    @Test
+    public void shouldHandleRangeWithMaxNullExclusive() {
+        //Given
+        Range range = rangeFactory.createRange(field, new byte[]{1}, true, null, false);
+
+        //When
+        Range canonicalisedRange = RangeCanonicaliser.canonicaliseRange(range);
+
+        //Then
+        assertThat(range.isInCanonicalForm()).isTrue();
+        assertThat(canonicalisedRange).isEqualTo(range);
+    }
+
+    @Test
+    public void shouldHandleRangeWithMaxNullInclusive() {
+        //Given
+        Range range = rangeFactory.createRange(field, new byte[]{1}, true, null, true);
+
+        //When
+        Range canonicalisedRange = RangeCanonicaliser.canonicaliseRange(range);
+
+        //Then;
+        assertThat(range.isInCanonicalForm()).isTrue();
+        assertThat(canonicalisedRange).isEqualTo(range);
+    }
+
+    private byte[] nextByteArrayValue(byte[] value) {
+        byte[] next = new byte[value.length + 1];
+        System.arraycopy(value, 0, next, 0, value.length);
+        next[value.length] = Byte.MIN_VALUE;
+        return next;
+    }
```

Comment:
This function is too similar to the production code. To know if the behaviour is correct, we should be able to see the actual value that's asserted just by reading the tests.

You can just put the literal value in the test. Just run it with an incorrect value in the assertion to see what the real one is.